addons/account/models/res_users.py

```diff
@@ -13,12 +13,13 @@ class Users(models.Model):
     def _check_one_user_type(self):
         super(Users, self)._check_one_user_type()
 
-        users_with_both_groups = self.filtered(lambda user:
-            user.has_group('account.group_show_line_subtotals_tax_included') and
-            user.has_group('account.group_show_line_subtotals_tax_excluded')
-        )
+        g1 = self.env.ref('account.group_show_line_subtotals_tax_included')
+        g2 = self.env.ref('account.group_show_line_subtotals_tax_excluded')
+
+        users_with_both_groups = self._fast_check_users_in_more_than_one_group([g1.id, g2.id])
+
         if users_with_both_groups:
-            names = ", ".join(users_with_both_groups.mapped('name'))
+            names = ", ".join(str(u[0]) for u in users_with_both_groups)
```

Comment:
Limit the size of the list.

Reply:
Do we really need to report the actual users? If we're editing a single user the ID is obvious. If we're editing groups, either we've just added a few users in it and it's obvious too, or we're touching "implied groups" and the list will be huge and useless anyway.

So I would simply omit it. Especially as there is a performance reason for doing so (see my comment on the query) ;-)